storage/transaction.go

```diff
@@ -50,10 +52,28 @@ func (s *Storage) unsafeGetTransactionWithBatchDetails(hash common.Hash) (
 func (s *TransactionStorage) GetTransactionsByCommitmentID(id models.CommitmentID) (models.GenericTransactionArray, error) {
 	batchedTxs := make([]stored.BatchedTx, 0, 32)
 
-	query := bh.Where("CommitmentID").Eq(id).Index("CommitmentID")
+	// BatchedTx are stored with CommitmentSlot as their primary key:
+	// - CommitmentSlot is (BatchID, IndexInBatch, IndexInCommitment)
+	// - CommitmentID   is (BatchId, IndexInBatch)
+	// This means that if a CommitmentID and a CommitmentSlot represent the same
+	// commitment then the serialization of the CommitmentID will be a prefix of the
+	// serialization of the CommitmentSlot. This makes it easy to look for all
+	// BatchedTxs for `id`
 
-	err := s.database.Badger.Find(&batchedTxs, query)
-	if err != nil {
+	// nolint: gocritic
+	seekPrefix := append(stored.BatchedTxPrefix, id.Bytes()...)
+
+	err := s.database.Badger.Iterator(seekPrefix, db.PrefetchIteratorOpts, func(item *bdg.Item) (bool, error) {
+		var batchedTx stored.BatchedTx
+		err := item.Value(batchedTx.SetBytes)
+		if err != nil {
+			return db.Continue, err
```

Comment:
This constant is a great idea, improves readability a lot 👏 Let's use it in all other places we Iterate over Badger keys. Maybe as a separate PR: https://linear.app/worldcoin/issue/PRO-240/use-dbcontinue-constant-in-all-iterator-usages.

Reply:
Opened #592 to do this as a followup PR